### ISSUE TITLE
Sofia_Sans_ММ.glyphs: fontmake build error

### ISSUE THREAD

**Issue by eliheuer** [lettersoup/Sofia-Sans#3]
When building `Sofia_Sans_ММ.glyphs` with [fontmake](https://github.com/googlefonts/fontmake), I get the following error: 
```
RecursionError: maximum recursion depth exceeded
```
This is a known [issue](https://github.com/googlefonts/glyphsLib/issues/375).

**Comment by eliheuer** [comment]
Full traceback:
```
fontmake -g Sofia_Sans_ММ.glyphs -o variable

INFO:fontmake.font_project:Building master UFOs and designspace from Glyphs source
INFO:glyphsLib.classes:Parsing "Sofia_Sans_ММ.glyphs" file into <GSFont>
Traceback (most recent call last):
  File "/Users/Air/Py/Venvs/type/bin/fontmake", line 10, in <module>
    sys.exit(main())
...
    notificationObj = Notification(notification, observableRef, data)
RecursionError: maximum recursion depth exceeded
```

**Comment by eliheuer** [comment]
A note on this from Marc:
```
there's something wrong with  zhe-cy
it may have components which reference one another so you get infinite recursion
I just threw a print statement in the function that kept getting repeated 
```

**Comment by eliheuer** [comment]
From the debugging tests I've been doing, it looks like the following glyphs have issues
```
zhe-cy
ka-cy
```
Removing these in GlyphsApp allows this file to build to UFO+Designspace files.

**Comment by lettersoup** [comment]
The zhe-cy, ka-cy should be fixed now.

**Comment by eliheuer** [comment]
@lettersoup Thanks the big issue now is the non-rectangular designspace: https://github.com/thundernixon/Encode-Sans/tree/master/docs/01-morphing-non-rect-designspace

**Comment by lettersoup** [comment]
To fix that I have to duplicate both Regular and both Bold masters. This will make the source file unnecessarily complicate (it will be useful just for the VF production). The math fix does not work because the instances do not match with the original ones (I´f already tried that). 
I sugest to send you a source file with duplicated masters just for the VF production and to keep the simple ones on Git. Please let us discuss it with @davelab6
<img width="178" alt="Bildschirmfoto 2019-08-20 um 18 20 19" src="https://user-images.githubusercontent.com/6229652/63365089-363cf580-c377-11e9-8002-1bc514c07b66.png">

**Comment by eliheuer** [comment]
Also, I just pulled commit https://github.com/lettersoup/Sofia-Sans/commit/04378f3ee67b2147a1f45de3fbc36fc8b7f66558 and I'm still having fontmake issues with zhe-cy.

Looking at the changelog, I see that only the date modified for each file was changed: https://github.com/lettersoup/Sofia-Sans/commit/04378f3ee67b2147a1f45de3fbc36fc8b7f66558

**Comment by eliheuer** [comment]
I'll make a new Issue for the non-rectangular designspace thing so we don't mix conversations.

Edit: here - https://github.com/lettersoup/Sofia-Sans/issues/6

**Comment by lettersoup** [comment]
> Also, I just pulled commit [04378f3](https://github.com/lettersoup/Sofia-Sans/commit/04378f3ee67b2147a1f45de3fbc36fc8b7f66558) and I'm still having fontmake issues with zhe-cy.
> 
> Looking at the changelog, I see that only the date modified for each file was changed: [04378f3](https://github.com/lettersoup/Sofia-Sans/commit/04378f3ee67b2147a1f45de3fbc36fc8b7f66558)

Both characters were a combination of a composite and an outline. They are now decomposed and with removed overlap. Masters are compatible. I  don't know what causes the issues.
<img width="1376" alt="Bildschirmfoto 2019-08-20 um 19 22 00" src="https://user-images.githubusercontent.com/6229652/63369266-da2a9f00-c37f-11e9-8ded-6de45d5d1db7.png">

**Comment by eliheuer** [comment]
Ok, that's great thanks. It just needs to be updated in the source files in Git, If I pull https://github.com/lettersoup/Sofia-Sans/commit/04378f3ee67b2147a1f45de3fbc36fc8b7f66558 it's still a combination of a composite and an outline.

**Comment by lettersoup** [comment]
Strange the files were updated. I just commit it again and also changed the outline starting point.
Would you please check it.
<img width="953" alt="Bildschirmfoto 2019-08-20 um 19 41 55" src="https://user-images.githubusercontent.com/6229652/63370504-a43aea00-c382-11e9-8dc6-7b9f57c06750.png">

**Comment by eliheuer** [comment]
Sure checking now, I'm still getting the infinite recursion error, even with the two glyphs decomposed and with removed overlap though.

**Comment by lettersoup** [comment]
Amm I dont have any idea what more could be fixed on the both characters. Do you have any suggestions?

**Comment by lettersoup** [comment]
Just cleared the mask layer zhe-cy, ka-cy. I dont think the mask causes the infinite recursion error, but would you please check again. Many thanks!

**Comment by eliheuer** [comment]
This is fixed as of commit 1488ab8cd6b64cd9539e5a0cc5f91df35ef52b45

Thanks @lettersoup!
